storage/innobase/xtrabackup/doc/source/version-check.rst

```diff
@@ -4,63 +4,113 @@
 
-Complex software projects (such as MySQL flavors) always have versions with
-minor or major compatibility problems, regressions and bugs, etc. It is too
-difficult to take care about all these compatibility nuances for a human, and
-this is the reason for a |version-check| feature to exist.
+Some Percona software contains “version checking” functionality which is a
+feature that enables Percona software users to be notified of available software
+updates to improve your environment security and performance. Alongside this,
+the version check functionality also provides Percona with information relating
+to which software versions you are running, coupled with the environment
+confirmation which the software is running within. This helps enable Percona to
+focus our development effort accordingly based on trends within our customer
+community.
 
-The feature allows compatible tools to query the Version Check database and
-print advice on any potential risks associated with any of the relevant
-software versions.
+
+The purpose of this document is to articulate the information that is
+collected, as well as to provide guidance on how to disable this functionality
+if desired.
 
 Usage
------
+=====
 
 *Version Check* was implemented in |pt| 2.1.4, and was enabled by default in
-version 2.2.1. Currently it is supported by most tools in |pt|, |pxb|, and
-|pmm|. In |Percona XtraBackup|, this feature was introduced in version 2.1.5. By
-default, it is also enabled.
+version 2.2.1. Currently it is supported as a ``--[no]version-check`` option
+by `a number of tools in Percona Toolkit <https://www.percona.com/doc/percona-toolkit/LATEST/genindex.html>`_,
+|pxb|, and |pmm|.
 
-Being called with Version Check enabled, the tool that supports this feature
-connects to a dedicated Percona server via a secure HTTPS channel. It checks
-its own version to query the server for possible updates, and also checks
+When launched  with Version Check enabled, the tool that supports this feature
+connects to a Percona's *version check service* via a secure HTTPS channel. It
+compares the locally installed version for possible updates, and also checks
 versions of the following software:
 
-* operating system
-* |Percona XtraBackup|
-* |MySQL|
+* Operating System
+* Percona Monitoring and Management (PMM)
+* MySQL
 * Perl
-* |MySQL| driver for Perl (DBD::mysql)
-* |pt|
+* MySQL driver for Perl (DBD::mysql)
+* Percona Toolkit
+
+Then it checks for and warns about versions with known problems if they are
+identified as running in the environment.
+
+Each version check request is logged by the server. Stored information consists
+of the checked system unique ID followed by the software name and version.
+The ID is generated either at installation or when the |version-check| query is
+submitted for the first time.
+
+.. note::
 
-Then, it checks for and warns about versions with known problems.
+   Prior to version 3.0.7 of |pt|, the system ID was calculated as an MD5 hash
```

Comment:
Version check script in xtrabackup is outdated and still using MD5(hostname)

Reply:
yes, this page is a copy of the same version check page from toolkit.